scene/main/viewport.cpp

```diff
@@ -1974,14 +1976,14 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 			return;
 		}
 
-		if (gui.key_focus && !gui.key_focus->is_visible_in_tree()) {
-			gui.key_focus->release_focus();
+		if (_gui_get_active_focus_control() && !_gui_get_active_focus_control()->is_visible_in_tree()) {
```

Comment:
Instead of calling the function twice, you should do something like
```C++
Control *focus_control = _gui_get_active_focus_control();
if (focus_control && ...
````